### PR TITLE
Fix for possible race for new block post consensus processing.

### PR DESCRIPTION
**Pull request by Frozen**
Fix for possible race for new block post consensus processing.

### CODE REVIEW

**Comment by sophoah** [comment]
this PR seems to break consensus, travis tests are  not moving
```
Waiting for localnet to boot...
Waiting for localnet to boot...
Localnet booted.
Waiting for epoch 2...
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

File: consensus/consensus_service.go [hunk 1]
```diff
@@ -620,14 +620,10 @@ func (consensus *Consensus) selfCommit(payload []byte) error {
 	return nil
 }
 
-// NumSignaturesIncludedInBlock returns the number of signatures included in the block
-func (consensus *Consensus) NumSignaturesIncludedInBlock(block *types.Block) uint32 {
+func (consensus *Consensus) numSignaturesIncludedInBlock(block *types.Block) uint32 {
 	count := uint32(0)
-	consensus.mutex.Lock()
```
Comment (sophoah) [inline consensus/consensus_service.go:626]: could you explain why we remove the lock here ?

Comment (sophoah) [reply consensus/consensus_service.go:626]: nevermind i saw it was already done before calling postprocessing function

**Comment by sophoah** [comment]
could you explain why this PR may fix the possible race condition with the post consensus process ?

**Comment by Frozen** [comment]
> could you explain why this PR may fix the possible race condition with the post consensus process ?
It removes https://github.com/harmony-one/harmony/pull/4677/files#diff-255a5cdd5516e7293e60a47a92503360676ddffb7e48e505598a589753b568c2L672 which execution is not predictable.

**Comment by Frozen** [comment]
This PR is unstable, i will close it

**Pull request closed without merging**